src/main/java/org/jenkinsci/plugins/workflow/support/steps/StageStepExecution.java

```diff
@@ -287,9 +287,11 @@ private static void cancel(StepContext context, StepContext newer) throws IOExce
          */
         void unblock(String message) {
             assert Thread.holdsLock(StageStepExecution.class);
-            assert waitingContext != null;
-            assert waitingBuild != null;
-            assert !holding.contains(waitingBuild);
+            assert waitingContext != null : message;
+            assert waitingBuild != null : message;
```

Comment:
Not sure, just throwing it in just in case. Anyway the failing assertion was the next line.